jax/_src/numpy/lax_numpy.py

```diff
@@ -4070,6 +4070,22 @@ def polyint(p, m=1, k=None):
     return true_divide(concatenate((p, k)), coeff)
 
 
+@_wraps(np.polydiv)
+def polydiv(u,v):
+  _check_arraylike("polydiv", u, v)
+  u, v = _promote_dtypes_inexact(u, v)
+  m = len(u) - 1
+  n = len(v) - 1
+  scale = 1. / v[0]
+  q = zeros(_max(m - n + 1, 1),)
+  for k in range(0, m-n+1):
+    d = scale * u[k]   
+    q = q.at[k].set(d)
+    u = u.at[k:k+n+1].add(-d*v)
+  u = u.at[~(abs(u) > 1E-14).cumsum().astype(bool)].set(0)
```

Comment:
Since this like is pretty non-trivial, I might factor out a helper function for clarity. Something like
```
def _zero_out_leading_elements(u, tol=1E-14):
  return u.at[~(abs(u) > 1E-14).cumsum().astype(bool)].set(0)
```
Also, the tolerance may have to be adjusted depending on the dtype of `u`. In numpy, I believe it's always cast to float64, so hard-coding 1E-14 is probably OK. In JAX, where possible we try to keep computations in 16 or 32 bit, matching the inputs, because these types have better support on some accelerators.